src/test/unit/libraries/AssetConfiguration.t.sol

```diff
@@ -0,0 +1,139 @@
+/** 
+    Created by Arcadia Finance
+    https://www.arcadia.finance
+
+    SPDX-License-Identifier: BUSL-1.1
+ */
+pragma solidity >=0.4.22 <0.9.0;
+
+import "../../../../lib/forge-std/src/Test.sol";
+import "../../../libraries/AssetConfiguration.sol";
+
+
+contract AssetConfigurationTest is Test {
+    using stdStorage for StdStorage;
+
+    uint8 public decimals = uint8(18);
+
+    address public nonCreator = address(1);
+    address public transmitter = address(32);
+    address public nonTransmitter = address(31);
+
+
+
+    function testSetCollateralFactor() public {
+        // Given: the initialConfig with all configuration parameters zero for the asset
+        AssetConfiguration.AssetDetailBitmap memory initialConfig = AssetConfiguration.AssetDetailBitmap({data: uint128(0)});
+
+        // When: The collateral factor is fetch from the config
+        uint256 collateralFactor1 = AssetConfiguration.getCollateralFactor(initialConfig);
+
+        // Then: It should be zero, because it is not set.
+        assertEq(collateralFactor1, 0);
+
+        // Then Given: The collateral ratio is set as 1
+        AssetConfiguration.setCollateralFactor(initialConfig, uint128(1));
+
+        // When: The collateral factor is fetched
+        uint256 collateralFactor2 = AssetConfiguration.getCollateralFactor(initialConfig);
+
+        // Then: New collateral factor should be 1 not 0
+        assertEq(collateralFactor2, 1);
+        assertTrue(collateralFactor2 != 0);
+
+    }
```

Comment:
@alikonuk1 an example for given, when, then testing comments

Reply:
I like it